File: source/Model/UserModel.cs
using TMS.Domain;

namespace TMS.Model;

public sealed record UserModel
{
    public long Id { get; init; }

    public string? UserName { get; init; }
    public string? NameAr { get; init; }
    public string? NameEn { get; init; }
    public string? MobileNumber { get; init; }

    public string? Email { get; init; }
    public Status Status { get; init; }
    public UserRole? Role { get; init; }

}

File: source/Web/Frontend/src/app/models/user.model.ts
import { AuthModel } from "./auth.model";
import {UserStatusEnum} from "./enum";
import {OptionModel} from "../components/select/option.model";

export class UserModel {
    id!: number;
    nameAr!: string;
    nameEn!: string;
    mobileNumber!: string;
    email!: string;
    auth!: AuthModel;
    status!:UserStatusEnum;
    department!: OptionModel;
    departmentId!: number;
    role!: OptionModel;
    userName! : string;
}

export class UpdatePasswordModel {
    id!: number;
    oldPassword!: string;
    newPassword!: string;
    confirmPassword!: string;
}
